modules/calib3d/src/fisheye.cpp

```diff
@@ -127,15 +127,27 @@ void cv::fisheye::projectPoints(InputArray objectPoints, OutputArray imagePoints
     {
         Vec3d Xi = objectPoints.depth() == CV_32F ? (Vec3d)Xf[i] : Xd[i];
         Vec3d Y = aff*Xi;
-        if (fabs(Y[2]) < DBL_MIN)
-            Y[2] = 1;
-        Vec2d x(Y[0]/Y[2], Y[1]/Y[2]);
+
+        if(fabs(Y[2]) < 1e-8)
+        {
+            if(Y[2] > 0)
+                Y[2] = 1e-8;
+            else
+                Y[2] = -1e-8;
+        }
+
+        Vec2d x(Y[0]/fabs(Y[2]), Y[1]/fabs(Y[2]));
 
         double r2 = x.dot(x);
         double r = std::sqrt(r2);
 
         // Angle of the incoming ray:
-        double theta = atan(r);
+        double theta;
+
+        if(Y[2] > 0)
+            theta = atan2(r, 1);
```

Comment:
this seems wrong to me. See https://april.eecs.umich.edu/media/pdfs/richardson2013iros.pdf Section II and the previous attempt on this https://github.com/opencv/opencv/pull/6801

Reply:
The main concern i see here is how mach atan2() precise? Is it works well with `DBL_MIN` or `1e-8`